attachment.go

```diff
@@ -71,14 +78,18 @@ func (c *Client) UploadAttachment(ctx context.Context, addrKR *crypto.KeyRing, r
 	return res.Attachment, nil
 }
 
-func (c *Client) getAttachment(ctx context.Context, attachmentID string) ([]byte, error) {
+func (c *Client) getAttachment(ctx context.Context, attachmentID string, buffer *bytes.Buffer) error {
 	res, err := c.doRes(ctx, func(r *resty.Request) (*resty.Response, error) {
 		return r.SetDoNotParseResponse(true).Get("/mail/v4/attachments/" + attachmentID)
 	})
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer res.RawBody().Close()
 
-	return io.ReadAll(res.RawBody())
+	buffer.Reset()
+
+	_, err = buffer.ReadFrom(res.RawBody())
+
+	return err
```

Comment:
```suggestion
	if _, err := buffer.ReadFrom(res.RawBody()); err != nil {
		return fmt.Errorf("failed to read attachment data: %w", err)
	}

	return nil
```

splitting the errors like this (and wrapping) could help diagnose issues in future

Reply:
Done